src/internal/Subject.ts

```diff
@@ -58,8 +61,10 @@ export class Subject<T> extends Observable<T> implements SubscriptionLike {
     errorContext(() => {
       this._throwIfClosed();
       if (!this.isStopped) {
-        const copy = this.observers.slice();
-        for (const observer of copy) {
+        if (!this.currentObservers) {
+          this.currentObservers = Array.from(this.observers);
```

Comment:
The general idea is we only clone when `currentObservers` is `null` (invalidated) and then we use `currentObservers` when possible.